standards/application/wakuchat.md

```diff
@@ -0,0 +1,172 @@
+
+The default inbox MUST be configured with the parameters:
+- **inbox_addr:** `client_address`
+
+One important property of the Inbox is that it can receive different payload types in a single location, by decoupling payload type from the associated contact_topic. New Conversation types will invariably create new payload types. Multiplexing payload types into a single inbox means the default inbox can be reused for undefined future conversationTypes.  
+
+The Default Inbox having public visibility does not preclude Clients having other Inboxes with different levels of visibility. 
+
+As the clients address is directly linked to Default Inbox, this pathway SHOULD only be used as a last resort.  
+
+
+## Envelopes
+As clients can support multiple sub-protocols simultaneously, a mechanism is required to determine how decode payloads.
+
+To process a  payload, a client must know how a payload was encoded/encrypted and what encryption state to use to decrypt it.
+
+**Encoding**
+One approach to tracking payload types is to encode it in the content topic. 
+Clients can then infer how to decode the payload, based on where it was received. 
+The downside is that only one type of payload can be received in a content topic. 
+Deploying new types requires a new content topic, which may go unnoticed to clients if they are running an older version or do not support the protocol. 
+Even if the client cannot read the payload, knowing it exists can signal to developers a need to update their applications to support the new types; reducing interaction failure and catching errors.
+
+**Encryption State**
+To determine which encryption state to use in decryption, applications must attach identifying information. One method to achieve this and ensure privacy, is  shield the required identifying information from public view using asymmetric encryption. However double encryption is expensive, and more opportunity for security faults.  
+
+
+### Conversation Hinting
+
+Conversation hinting is a mechanism to provide recipients with the necessary information, without exposing conversational metadata.  
+
+Naively, conversation_ids are sufficient to allow clients to determine how to decrypt a payload. As the conversation_id corresponds to the ConversationType which defines how payloads are encdoded, and the encryption state is linked to the specific identifier. However this would leak Conversation metadata, as observers could determine which payloads belonged together into a group. Instead a hint is used which allows clients to check if this payload is of interest. A ConversationHint does not contain identifying information. 
+
+ConversationHints are computed by using a salted hash of the `conversationId`. specifically defined as `lowercase_hex(blake2s(salt || conversation_id))`.
```

Comment:
This approach to hinting offers insufficient privacy protections. A strategy using either SealedSender or Ratcheting Private Identifiers would be better suited.